### PR TITLE
[DEVOPS-365] Modifications for slightly improved dockerfile.

### PR DESCRIPTION
**Pull request by jmfadeley**
#### What's this PR do?
Slight cleanup of the Dockerfile and some seeding options that weren't there before. Chiefly, the ability to give a password for the seeding admin, but please only give this via the command line. 

#### Related JIRA tickets:
DEVOPS-365.

#### How should this be manually tested?
Create a docker PG instance you can setup:
`docker run --name auth-db --network indaba-network -e POSTGRES_USER=amida -e POSTGRES_PASSWORD=amida -e POSTGRES_DB=amida-auth-microservice postgres:9.6`

Run a build:
`docker build -t amida-auth-microservice .`

Then deploy, note the password: 
`docker run -p 4000:4000 --name amida-auth-microservice --network indaba-network -v C:\Users\Odin\workspace\amida-auth-microservice\.env:/app/.env -e AUTH_SERVICE_PG_HOST=auth-db -e AUTH_SERVICE_SEED_ADMIN_PASSWORD=example  -e AUTH_SERVICE_PG_PASSWORD=amida -e AUTH_SERVICE_PG_USER=amida amida-auth-microservice yarn migrate && yarn seed`

From here, you can use a postman call to the following address with the following criteria to "login" as the admin:
`http://localhost:4000/api/v1/auth/login`
POST
`username: admin`
`email: admin`
`password: example`

#### Screenshots (if appropriate):

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,36 +1,20 @@
...
+RUN yarn install --pure-lockfile && \
+    yarn build && \
+    yarn install --production --frozen-lockfile
 
 # expose port 4000
 EXPOSE 4000
 
 # cmd to start service
-CMD [ "node", "dist/index.js"]
+CMD yarn migrate && yarn serve
```
Comment (rmharrison) [inline Dockerfile]: The *Dockerfile* proper should probably only `yarn serve`

File: Dockerfile [hunk 2]
```diff
@@ -1,36 +1,20 @@
-# take default image of node boron i.e  node 6.x
-FROM node:8.9 as builder
-RUN npm i -g yarn
-
-# create app directory in container
-RUN mkdir -p /app
+# node image
+FROM node:8.10.0
```
Comment (rmharrison) [inline Dockerfile]: 2-cents: Could use a lighter base image.
Alpine is about 1/10 the size (~70 Mb vs 700 Mb).

File: package.json [hunk 3]
```diff
@@ -12,6 +12,7 @@
   },
   "scripts": {
     "start": "gulp serve",
+    "serve": "node dist/index.js",
```
Comment (rmharrison) [inline package.json]: Good catch + Yas! :smile:

File: Dockerfile [hunk 4]
```diff
@@ -1,36 +1,20 @@
...
-WORKDIR /app
-
-COPY --from=builder /app /app
-
 # set up public and private keys
 RUN echo -e 'y\n'|ssh-keygen -q -t rsa -b 4096 -N "" -f private.key &&\
     openssl rsa -in private.key -pubout -outform PEM -out private.key.pub
 
-#USER 50000:50000
+RUN yarn install --pure-lockfile && \
```
Comment (rmharrison) [inline Dockerfile]: Reduces image size by containing to one slice. :+1:

File: Dockerfile (hunk 1 above)
Comment (jmfadeley) [reply Dockerfile]: Should we run `yarn migrate` as part of `yarn serve` then? I was really kind of hoping to have a self-upgrading serve.

File: Dockerfile (hunk 4 above)
Comment (jmfadeley) [reply Dockerfile]: As in, have it all one line?

File: Dockerfile [hunk 5]
```diff
@@ -1,36 +1,21 @@
...
+# Obtain ssh-keygen and patch for snyk
+RUN apk update && apk add --no-cache openssh-keygen && apk add --no-cache openssl && apk add --no-cache patch
 
 # set up public and private keys
-RUN echo -e 'y\n'|ssh-keygen -q -t rsa -b 4096 -N "" -f private.key &&\
+RUN echo -e 'y\n'|ssh-keygen -q -t rsa -b 4096 -N "" -f private.key && \
     openssl rsa -in private.key -pubout -outform PEM -out private.key.pub
 
-#USER 50000:50000
+RUN yarn install --pure-lockfile && yarn build && yarn install --production --frozen-lockfile
```
Comment (rmharrison) [inline Dockerfile]: Using the [multi-stage build](https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds) means that we no longer need to include the install/build/install on the same line.

**Comment by jmfadeley** [comment]
@rmharrison Okay, changes now include using alpine, multi-stage build, the auth integration test, and I've removed the `yarn migrate`. I think that should do it.

**Comment by jmfadeley** [comment]
Also, note that I've changed the command line for testing purposes to provide the same username and password as well as the `yarn migrate`, but that the Dockerfile does not migrate within itself.

File: create_db.js [hunk 6]
```diff
@@ -9,13 +9,13 @@ pg.connect(conStringPri, (err, client, done) => { // eslint-disable-line no-unus
         process.exit(1);
     }
     // create the db and ignore any errors, for example if it already exists.
-    client.query(`CREATE DATABASE ${config.postgres.db};`, (err1) => { // eslint-disable-line no-unused-vars
+    client.query(`CREATE DATABASE "${config.postgres.db}";`, (err1) => { // eslint-disable-line no-unused-vars
```
Comment (rmharrison) [inline create_db.js]: YMMV. :man_shrugging: 

Avoids the string interpolation with \`
`client.query('CREATE DATABASE "$1"', config.postgres.db, ...)`

Comment (rmharrison) [inline create_db.js]: Typo: `err1` --> `err`

Comment (jmfadeley) [reply create_db.js]: We took a look into this and it didn't work. :./

**Pull request merged**